### PR TITLE
Lang discovery adjustments

### PR DESCRIPTION
**Pull request by unho**

### CODE REVIEW

**Comment by phlax** [comment]
tested locally and this fixes a few issues. Needs some tests tho. Gtm once test/s added

File: tests/i18n/override.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    # Test header with unsupported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='FAIL')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+
+@pytest.mark.django_db
+def test_get_lang_from_session(rf, client):
+    # Test no session.
+    request = rf.get("")
+    assert not hasattr(request, 'session')
```
Comment (phlax) [inline tests/i18n/override.py]: i think this assert is not what is being tested - better add a comment to say that there is no session

File: tests/i18n/override.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'en-US'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) == 'en-US'
+
+    # Test cookie with longer underscore language code for a supported
+    # language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'gl_ES'
```
Comment (phlax) [inline tests/i18n/override.py]: missing assert ?

File: tests/i18n/override.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    # Test cookie with longer hyphen language code for a supported language.
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'fr-FR'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
```
Comment (phlax) [inline tests/i18n/override.py]: should this not fallback to `fr` ?

**Comment by phlax** [comment]
im not seeing in any of the current tests anything testing for the fallback codes that we are adding. Apart from seeing the fallback working, it would also be good to see something where it doesnt fallback - eg `spam-EGGS`

File: tests/i18n/override.py (hunk 2 above)
Comment (unho) [reply tests/i18n/override.py]: No, here we are telling the session to use this particular language. Maybe there is a better way to achieve this, but this works.

File: tests/i18n/override.py (hunk 3 above)
Comment (unho) [reply tests/i18n/override.py]: Probably yes, but will need to change the function to support that. Will look into it.

File: tests/i18n/override.py (hunk 1 above)
Comment (unho) [reply tests/i18n/override.py]: Will do

File: tests/i18n/override.py (hunk 2 above)
Comment (unho) [reply tests/i18n/override.py]: It results that the missing assert is a few lines below.

File: tests/i18n/override.py (hunk 3 above)
Comment (unho) [reply tests/i18n/override.py]: Updated PR to handle this.

File: tests/i18n/override.py (hunk 2 above)
Comment (unho) [reply tests/i18n/override.py]: Fixed now.

**Comment by unho** [comment]
Updated with all raised concerns.

**Comment by phlax** [comment]
lgtm

**Pull request merged**